presto-hive/src/main/java/com/facebook/presto/hive/security/ranger/RangerBasedAccessControl.java

```diff
@@ -0,0 +1,516 @@
+    }
+
+    /**
+     * Check if identity is allowed to add columns to the specified table in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanAddColumn(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName)
+    {
+        if (!checkAccess(identity, tableName, null, HiveAccessType.ALTER)) {
+            denyAddColumn(tableName.toString(), format("Access denied - User [ $s ] does not have is not have [ALTER] " +
+                    "privilege on [ %s/%s ] ", identity.getUser(), tableName.getSchemaName(), tableName.getTableName()));
+        }
+    }
+
+    /**
+     * Check if identity is allowed to drop columns from the specified table in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanDropColumn(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName)
+    {
+        if (!checkAccess(identity, tableName, null, HiveAccessType.ALTER)) {
+            denyDropColumn(tableName.toString(), format("Access denied - User [ $s ] does not have is not have [ALTER] " +
+                    "privilege on [ %s/%s ] ", identity.getUser(), tableName.getSchemaName(), tableName.getTableName()));
+        }
+    }
+
+    /**
+     * Check if identity is allowed to rename a column in the specified table in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanRenameColumn(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName)
+    {
+        if (!checkAccess(identity, tableName, null, HiveAccessType.ALTER)) {
+            denyRenameColumn(tableName.toString(), format("Access denied - User [ $s ] does not have is not have [ALTER] " +
+                    "privilege on [ %s/%s ] ", identity.getUser(), tableName.getSchemaName(), tableName.getTableName()));
+        }
+    }
+
+    /**
+     * Check if identity is allowed to select from the specified columns in a relation.  The column set can be empty.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanSelectFromColumns(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName, Set<String> columnNames)
+    {
+        Set<String> deniedColumns = new HashSet<>();
+        for (String column : columnNames) {
+            if (!checkAccess(identity, tableName, column, HiveAccessType.SELECT)) {
+                deniedColumns.add(column);
+            }
+        }
+        if (deniedColumns.size() > 0) {
+            denySelectColumns(tableName.toString(), deniedColumns, format("Access denied - User [ $s ] does not have is not have [ALTER] " +
```

Comment:
lets edit the error message to be in sync with what I stated above and also mention the columns that the user does not have access to

Reply:
**Changing tableName.toString() to tableName.getTableName()** as not all callback methods provides string schemaName arg but provides SchemaTableName wrapper class